### PR TITLE
[DUOS-2250][risk=no] NPE Fix

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2250

Fixes NPE seen here: https://sentry.io/organizations/broad-institute/issues/3882456746/?project=153297&query=is%3Aunresolved&referrer=issue-stream

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java [hunk 1]
```diff
@@ -88,8 +88,8 @@ public void addApprovedDARLine(FileWriter darWriter, Election election, DataAcce
 
     public void addReviewedDARLine(FileWriter darWriter, Election election, DataAccessRequest dar, String darCode, String consentName, String translatedUseRestriction) throws IOException {
         String finalVote = election.getFinalVote() ? "Yes" : "No";
-        String content2 = formatTimeToDate(election.getFinalVoteDate().getTime()) + DEFAULT_SEPARATOR +
-                          finalVote;
+        String electionDate = (Objects.nonNull(election.getFinalVoteDate())) ? formatTimeToDate(election.getFinalVoteDate().getTime()) : "";
+        String content2 = electionDate + DEFAULT_SEPARATOR + finalVote;
```
Comment (connorlbark) [inline src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java]: what does this variable do? maybe needs more descriptive name

File: src/test/java/org/broadinstitute/consent/http/service/DataAccessReportsParserTest.java [hunk 2]
```diff
@@ -18,27 +13,28 @@
...
-            "Notes:\n" +
-            "Future use for methods research (analytic/software/technology development) is not prohibited\n" +
-            "Future use as a control set for diseases other than those specified is not prohibited";
+    private final String sDUL = """
+        Samples Restricted for use with "cancer" [DOID_162(CC)]
+        Future use by for-profit entities is prohibited [NPU]
+        Future use of aggregate-level data for general research purposes is prohibited [NPNV]
+        Notes:
+        Future use for methods research (analytic/software/technology development) is not prohibited
+        Future use as a control set for diseases other than those specified is not prohibited""";
```
Comment (connorlbark) [inline src/test/java/org/broadinstitute/consent/http/service/DataAccessReportsParserTest.java]: nit: """ should be on newline

Comment (connorlbark) [reply src/test/java/org/broadinstitute/consent/http/service/DataAccessReportsParserTest.java]: also: unrelated, but should `not prohibited` be `allowed`? confusing verbage

File: src/test/java/org/broadinstitute/consent/http/service/DataAccessReportsParserTest.java [hunk 3]
```diff
@@ -107,29 +105,71 @@ public void testDataAccessReviewedReport() throws IOException {
...
             if(i == 0) {
-                assertTrue(columns[0].equals(HeaderDAR.DAR_ID.getValue()));
-                assertTrue(columns[1].equals(HeaderDAR.DATASET_NAME.getValue()));
-                assertTrue(columns[2].equals(HeaderDAR.DATASET_ID.getValue()));
-                assertTrue(columns[3].equals(HeaderDAR.CONSENT_ID.getValue()));
-                assertTrue(columns[4].equals(HeaderDAR.CODED_VERSION_SDUL.getValue()));
-                assertTrue(columns[5].equals(HeaderDAR.CODED_VERSION_DAR.getValue()));
-                assertTrue(columns[6].equals(HeaderDAR.DATE_REQUEST_APPROVAL_DISAPROVAL.getValue()));
-                assertTrue(columns[7].equals(HeaderDAR.APPROVED_DISAPPROVED.getValue()));
+                assertEquals(columns[0], HeaderDAR.DAR_ID.getValue());
```
Comment (connorlbark) [inline src/test/java/org/broadinstitute/consent/http/service/DataAccessReportsParserTest.java:153]: i like the assertEquals cleanup here

**Review by connorlbark — APPROVED** [review]
a couple minor nits, but with the added test i'd say it is looking good functionally! also, i don't know if this is a common code path, but as an aside, i find the general structure of this code very confusing. i know it's working with a TSV so the structure is inherently confusing with `columns[i]` for example - but maybe we do a bit of cleanup by keeping the indices of the columns as constants, e.g. `int ORGANIZATION_IDX = 5`. either way, out of the scope of this PR and this fixes what you wanted to fix, so 👍

File: src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java]: `addDARLine` has two custom content variables that it uses - this is the second one :-) Will rename.

File: src/test/java/org/broadinstitute/consent/http/service/DataAccessReportsParserTest.java (hunk 2 above)
Comment (rushtong) [reply src/test/java/org/broadinstitute/consent/http/service/DataAccessReportsParserTest.java]: > also: unrelated, but should `not prohibited` be `allowed`? confusing verbage

Language choices here are unrelated to the PR. However, these are also intentional. I believe I have made the same arguments before. [See this search for examples](https://github.com/DataBiosphere/consent-ontology/search?q=%22Future+use+as+a+control+set%22) of why this test does what it does.

Comment (rushtong) [reply src/test/java/org/broadinstitute/consent/http/service/DataAccessReportsParserTest.java]: This is a better link for the `not prohibited` language: https://github.com/DataBiosphere/consent-ontology/blob/484f91430e08513817dd4ed60e999e598466bdee/src/main/java/org/broadinstitute/dsde/consent/ontology/datause/services/TextTranslationServiceImpl.java#L48

**Pull request merged**